ibm_vpd_utils.cpp

```diff
@@ -700,17 +701,30 @@ std::string getPrintableValue(const std::variant<Binary, std::string>& kwVal)
     std::string kwString{};
     std::visit(
         [&kwString](auto&& kwVal) {
-            const auto it =
-                std::find_if(kwVal.begin(), kwVal.end(),
-                             [](const auto& kw) { return !isprint(kw); });
-            if (it != kwVal.end())
+        const auto it =
+            std::find_if(kwVal.begin(), kwVal.end(),
+                         [](const auto& kw) { return !isprint(kw); });
+        if (it != kwVal.end())
+        {
+            bool printable = true;
+            for (auto itr = it; itr != kwVal.end(); itr++)
             {
-                kwString = hexString(kwVal);
+                if (*itr != 0x00)
+                {
+                    kwString = hexString(kwVal);
+                    printable = false;
+                    break;
+                }
             }
-            else
+            if (printable)
             {
-                kwString = std::string(kwVal.begin(), kwVal.end());
+                kwString = std::string(kwVal.begin(), it);
             }
+        }
+        else
+        {
+            kwString = std::string(kwVal.begin(), kwVal.end());
+        }
```

Comment:
Hi Andrew,
These changes have reverted to old one, so the defect may not be fixed. Can you please change it to correct one, if possible. Thanks.

Reply:
Not sure I understand @GiridhariKrishnan , this clang format run was based on what was in 1050 when I ran it. Did you have a change come in after that changed this code in some way? If so, a merge conflict should have come for one of us depending on who merged first (and then we'd be forced to fix it).